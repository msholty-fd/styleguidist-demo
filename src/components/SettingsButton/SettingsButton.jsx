import React from 'react'
import PropTypes from 'prop-types'
import IconButton from 'material-ui/IconButton'
import Settings from 'material-ui/svg-icons/action/settings'
import ArrowDropDown from 'material-ui/svg-icons/navigation/arrow-drop-down'
import getMuiTheme from 'material-ui/styles/getMuiTheme'
import theme from '../../styles/Theme'
import { Theme } from '../../styles'
import colors from '../../styles/Colors'

const styles = {
  button: {
    width: 50,
    height: 27,
    marginTop: 7,
    padding: 7,
    background: colors.basicButton,
    borderRadius: 3
  },
  icon: {
    width: 14,
    height: 14,
    color: Theme.palette.alternateTextColor
  }
}

export default class SettingsButton extends React.Component {
  static childContextTypes = {
    muiTheme: PropTypes.object.isRequired
  }

  getChildContext () {
    return {muiTheme: getMuiTheme(theme)}
  }

  render () {
    return (
      <IconButton
        iconStyle={styles.icon}
        style={styles.button}
        disableTouchRipple
      >
        <Settings />
        <ArrowDropDown />
      </IconButton>
    )
  }
}
