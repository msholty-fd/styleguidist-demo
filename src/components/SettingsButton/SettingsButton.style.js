import { Theme } from 'styles'
import colors from 'styles/Colors'

export default {
  button: {
    width: 50,
    height: 27,
    marginTop: 7,
    padding: 7,
    background: colors.basicButton,
    borderRadius: 3
  },
  icon: {
    width: 14,
    height: 14,
    color: Theme.palette.alternateTextColor
  }
}
