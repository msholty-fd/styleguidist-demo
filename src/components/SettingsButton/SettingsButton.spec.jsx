import React from 'react'
import renderer from 'react-test-renderer'
import SettingsButton from './SettingsButton'
import { describe, it, expect } from 'global'

jest.mock('material-ui/IconButton', () => 'IconButton')
jest.mock('material-ui/svg-icons/action/settings', () => 'Settings')
jest.mock('material-ui/svg-icons/navigation/arrow-drop-down', () => 'ArrowDropDown')

const component = <SettingsButton />

describe('<SettingsButton />', () => {
  it('should match the snapshot', () => {
    const renderedComponent = renderer.create(component)
    const tree = renderedComponent.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
