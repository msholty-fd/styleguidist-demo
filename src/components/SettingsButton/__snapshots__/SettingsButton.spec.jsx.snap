// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SettingsButton /> should match the snapshot 1`] = `
<IconButton
  disableTouchRipple={true}
  iconStyle={
    Object {
      "color": "#ffffff",
      "height": 14,
      "width": 14,
    }
  }
  style={
    Object {
      "background": "#a8b0b0",
      "borderRadius": 3,
      "height": 27,
      "marginTop": 7,
      "padding": 7,
      "width": 50,
    }
  }
>
  <Settings />
  <ArrowDropDown />
</IconButton>
`;
